### DISCUSSION TITLE
Error   | Quadraturefree ADER-DG is only implemented for tetrahedral and hexahedral.

### DISCUSSION THREAD

**Discussion by smallriver666** [SeisSol/SeisSol#846] [Q&A]
hello, when I run SeisSol I meet this problem  `Error| Quadraturefree ADER-DG is only implemented for tetrahedral and hexahedral.`, does this caused by ill tetrahedral,because when I generate mesh, I meet this warning.
![image](https://user-images.githubusercontent.com/104721358/234213717-72054ae0-ba36-4aaf-ab3a-534fafb90e69.png)

**Comment by krenzland** [comment]
Well, what is the warning that you're getting?

**Reply by smallriver666** [reply]
> Well, what is the warning that you're getting?

my warning is `Warning : 108 ill-shaped tets are still in the mesh` when I use gmsh to generate mesh.

**Comment by davschneller** [comment]
The error "Quadraturefree ADER-DG[...]" means that SeisSol has effectively (but successfully) read in an empty mesh.

**Reply by smallriver666** [reply]
> The error "Quadraturefree ADER-DG[...]" means that SeisSol has effectively (but successfully) read in an empty mesh.

Empty mesh can stop my SeisSol run, how can I find where is the empty mesh or how to prevent it ?

**Reply by davschneller** [reply]
Can you supply us with the full log from the meshing process? In particular: what does PUMgen output? How big is the resulting mesh file (it has the ending `.puml.h5`)? What do get if you run `h5dump -pH meshfile.puml.h5` (with `meshfile.puml.h5` being your meshfile name)?

**Reply by smallriver666** [reply]
> Can you supply us with the full log from the meshing process? In particular: what does PUMgen output? How big is the resulting mesh file (it has the ending `.puml.h5`)? What do get if you run `h5dump -pH meshfile.puml.h5` (with `meshfile.puml.h5` being your meshfile name)?

This is `h5dump -pH meshfile.puml.h5` [output](https://github.com/smallriver666/MeshError/blob/main/h5dump%20-pH%20meshfile.puml.h5). This one is [meshing process ](https://github.com/smallriver666/MeshError/blob/main/meshing%20process).

**Reply by davschneller** [reply]
Hi, sorry for the delay with replying. ... It the problem by now still persists: what does/did the log for SeisSol look like before the error?

**Reply by smallriver666** [reply]
> Hi, sorry for the delay with replying. ... It the problem by now still persists: what does/did the log for SeisSol look like before the error?

It is a strange problem, I have regenerate the grid, the problem have solved.

**Comment by smallriver666** [comment]
I found out why this error occurs, I encountered this problem when I simulated an example with only 10 million meshes using `nodes*cores=143*7`, and I did not encounter this problem when I used `nodes*cores=27*4`.

So I suspect that this is due to a problem with mesh partitioning caused by using many CPU cores to compute with a small number of meshes.
